address/book.go

```diff
@@ -327,15 +332,31 @@ func (b *Book) NextScriptKey(ctx context.Context,
 func (b *Book) ListAddrs(ctx context.Context,
 	params QueryParams) ([]AddrWithKeyInfo, error) {
 
-	return b.cfg.Store.QueryAddrs(ctx, params)
+	addrs, err := b.cfg.Store.QueryAddrs(ctx, params)
+	if err != nil {
+		return nil, err
+	}
+
+	// Add the proof courier address to all Tap addresses.
+	for i := range addrs {
+		addrs[i].Tap.ProofCourierAddr = b.cfg.ProofCourierAddr
```

Comment:
I think we should check if the courier address is empty before overwriting it? Otherwise you would never see a custom address if you set one explicitly.

Reply:
Same in all calls below in this file.

Reply:
Ah, we're missing the persistence to the DB in this PR, that's why this was added. Actually I think the diff would be smaller by adding the new DB field.

Reply:
Yeah I think the addrs should have this on disk. Then this way we avoid "downgrades" due to someone modifying the default commit.